### PR TITLE
Updated listOVPN to Include Expiration Dates

### PR DESCRIPTION
**Pull request by psgoundar**
Updated listOVPN.sh script to include Expiration Dates and Cleaned up Escape Sequences.

### CODE REVIEW

File: scripts/openvpn/listOVPN.sh [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
         exit 1
 fi
 
 printf ": NOTE : The first entry should always be your valid server!\n"
-printf "\n"
-printf "\e[1m::: Certificate Status List :::\e[0m\n"
-printf " ::\e[4m  Status  \e[0m||\e[4m   Name   \e[0m:: \n"
+printf "\\n"
+printf "\\e[1m::: Certificate Status List :::\\e[0m\\n"
+printf "\\e[4mStatus\\e[0m ::  \\e[4mName\\e[0m\\e[0m        ::  \\e[4mExpiration \\e[0m\\n"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:15]:
```suggestion
{
printf "\\e[4mStatus\\e[0m  \t  \\e[4mName\\e[0m\\e[0m  \t  \\e[4mExpiration\\e[0m\\n"
```

File: scripts/openvpn/listOVPN.sh [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
 
 while read -r line || [ -n "$line" ]; do
     STATUS=$(echo "$line" | awk '{print $1}')
-    NAME=$(echo "$line" | sed -e 's:.*/CN=::')
+    NAME=$(echo "$line" | awk '{print $5}' | awk -FCN= '{print $2}')
+    EXPD=$(echo "$line" | awk '{if (length($2) == 15) print $2; else print "20"$2}' | cut -b 1-8 | date +"%b %d %Y" -f -)
+        
     if [ "${STATUS}" == "V" ]; then
-        printf "     Valid   ::   %s\n" "$NAME"
+        printf "     Valid   ::   %s :: %s\\n" "$NAME" "$EXPD"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:23]:
```suggestion
        printf "Valid  \t  %s  \t  %s\\n" "$NAME" "$EXPD"
```

File: scripts/openvpn/listOVPN.sh [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
     elif [ "${STATUS}" == "R" ]; then
-        printf "     Revoked ::   %s\n" "$NAME"
+        printf "     Revoked   ::   %s :: %s\\n" "$NAME" "$EXPD"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:25]:
```suggestion
        printf " Revoked  \t  %s  \t  %s\\n" "$NAME" "$EXPD"
```

File: scripts/openvpn/listOVPN.sh [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
     else
-        printf "     Unknown ::   %s\n" "$NAME"
+        printf "     Unknown   ::   %s :: %s\\n" "$NAME" "$EXPD"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:27]:
```suggestion
        printf "Unknown  \t  %s  \t  %s\\n" "$NAME" "$EXPD"
```

File: scripts/openvpn/listOVPN.sh [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
     fi
-done <${INDEX}
-printf "\n"
+
+done <${INDEX} | column -t
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:30]:
```suggestion
done <${INDEX}
printf "\\n"
} | column -t -s $'\t'
```

File: scripts/openvpn/listOVPN.sh [hunk 6]
```diff
@@ -1,27 +1,31 @@
...
+printf "\\n"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:31]:
```suggestion

```

File: scripts/openvpn/listOVPN.sh [hunk 7]
```diff
@@ -1,27 +1,31 @@
 #!/usr/bin/env bash
 # PiVPN: list clients script
+# Updated Script to include Expiration Dates and Clean up Escape Seq -- psgoundar
 
 INDEX="/etc/openvpn/easy-rsa/pki/index.txt"
-printf "\n"
+printf "\\n"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:6]:
```suggestion

```

**Comment by orazioedoardo** [comment]
Your pull request made the alignment a bit off, so I suggested some small changes to fix that, and also make the output more consistent with `listCONF.sh`.

```

: NOTE : The first entry should always be your valid server!

::: Certificate Status List :::
Status ::  Name        ::  Expiration 
Valid  ::  debian_9ab07e1d-9ce8-4a0a-993d-e24c63816ed3  ::  mar  30  2030
Valid  ::  user                                         ::  mar  17  2023

```

```
: NOTE : The first entry should always be your valid server!

::: Certificate Status List :::
Status      Name                                             Expiration
Valid       debian_9ab07e1d-9ce8-4a0a-993d-e24c63816ed3      mar 30 2030
Valid       user                                             mar 17 2023
```

(the underline and bold text are still there)

**Comment by psgoundar** [comment]
Thanks orazioedoardo for the review. I've commited the  changes that you recommended.

File: scripts/openvpn/listOVPN.sh [hunk 8]
```diff
@@ -12,20 +11,23 @@ fi
...
     STATUS=$(echo "$line" | awk '{print $1}')
     NAME=$(echo "$line" | awk '{print $5}' | awk -FCN= '{print $2}')
     EXPD=$(echo "$line" | awk '{if (length($2) == 15) print $2; else print "20"$2}' | cut -b 1-8 | date +"%b %d %Y" -f -)
         
     if [ "${STATUS}" == "V" ]; then
-        printf "     Valid   ::   %s :: %s\\n" "$NAME" "$EXPD"
+        printf "Valid  \t  %s  \t  %s\\n" "$NAME" "$EXPD"
     elif [ "${STATUS}" == "R" ]; then
-        printf "     Revoked   ::   %s :: %s\\n" "$NAME" "$EXPD"
+        printf " Revoked  \t  %s  \t  %s\\n" "$NAME" "$EXPD"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:25]:
```suggestion
        printf "Revoked  \t  %s  \t  %s\\n" "$NAME" "$EXPD"
```

File: scripts/openvpn/listOVPN.sh [hunk 9]
```diff
@@ -12,20 +11,23 @@ fi
...
+        printf "Unknown  \t  %s  \t  %s\\n" "$NAME" "$EXPD"
     fi
 
-done <${INDEX} | column -t
-printf "\\n"
\ No newline at end of file
+done <${INDEX}
+printf "\\n"
+} | column -t -s $'\t'
+printf "\\n"
```
Comment (orazioedoardo) [inline scripts/openvpn/listOVPN.sh:33]:
```suggestion
```

**Comment by orazioedoardo** [comment]
> Thanks orazioedoardo for the review. I've commited the changes that you recommended.

I added two minor changes because of a typo in previous suggestions.

**Pull request merged**